src/arithmetic.jl

```diff
@@ -572,6 +575,19 @@ for T in (:Taylor1, :TaylorN)
     end
 end
 
+# in-place product: `a` <- `a*b`
+# this method computes the product `a*b` and saves it back into `a`
+# assumes `a` and `b` are of same order
+function mul!(a::TaylorN{T}, b::TaylorN{T}) where {T<:Number}
+    for k in reverse(eachindex(a))
+        mul!(a, a, b[0][1], k)
+        for l in 1:k
+            mul!(a[k], a[k-l], b[l])
+        end
+    end
+    return nothing
+end
+
```

Comment:
Not sure if this is a good proposal, but here it goes: the code for the next method of `mul!` could be simplified using the new one proposed in this PR, e.g., by copying `a` into `res`, and then using `mul!(res, b)`.